sqldelight-gradle-plugin/src/main/kotlin/com/squareup/sqldelight/gradle/SqlDelightTask.kt

```diff
@@ -51,13 +51,15 @@ open class SqlDelightTask : SourceTask() {
   @TaskAction
   fun generateSqlDelightFiles() {
     outputDirectory?.deleteRecursively()
-    val environment = SqlDelightEnvironment(
-        sourceFolders = sourceFolders.filter { it.exists() },
-        dependencyFolders = dependencySourceFolders.filter { it.exists() },
-        properties = properties,
-        outputDirectory = outputDirectory,
-        moduleName = project.name.filter { it.isLetter() }
-    )
+    val environment = synchronized(environmentLock) {
+      SqlDelightEnvironment(
```

Comment:
The static state side-effects from this object creation are somewhat depressing. I'm also not sure whether it leaks or not, but it kinda seems like it does.

Reply:
yea, its mostly out of our control. Our type (`SqlDelightEnvironment`) doesnt have static state, but the intellij type `ApplicationCoreEnvironment` does which is what causes the problem.

Theres probably a much more elegant solution to this problem where we create a single environment in memory and refactor the psi layer to allow passing files in ad-hoc instead of during initialization, I'm just not super enthused to go down that path right now compared to other stuff for sqld. Happy to add a TODO which says that tho